terraform/setup/setup.tf

```diff
@@ -0,0 +1,150 @@
+# in this module, we create the necessary common resources for the integ-tests, this setup module will only need to be executed once.
+# vpc, iam role, security group, the number of those resources could be limited, creating them concurrently for every pr would trigger throttling issue.
+
+provider "aws" {
+  region = var.region
+}
+
+# create ssh key pair and upload them to s3
+resource "tls_private_key" "ssh_key" {
```

Comment:
Ah I see you generate key here - do you need to store it in a bucket? I think it's ok to generate key ever PR run, unless you know of a throttling limitation of `aws_key_pair`. Less permanent keys is safer :)

Reply:
yeah, just keep the key in s3 for debug purpose. And there might be a limit on it, but not like vpc (5 vpcs in one account), it could be hard to reach.